### PR TITLE
chore(ci): replace deprecated toolchain action

### PR DESCRIPTION
**Pull request by The-personified-devil**
This would also allow us to directly specify a version of rust that we want

### CODE REVIEW

**Comment by zarik5** [comment]
Why is this PR draft?

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -164,13 +141,10 @@ jobs:
...
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: stable
-          override: true
-      - uses: Swatinem/rust-cache@v1
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@1.74
```
Comment (zarik5) [inline .github/workflows/rust.yml:145]: I'd say not to download the toolchain for the current msrv. It's going to be done by xtask and I want the version to be pulled dynamically from the root Cargo.toml.

**Review by zarik5 — CHANGES_REQUESTED** [review]

File: .github/workflows/rust.yml (hunk 1 above)
Comment (The-personified-devil) [reply .github/workflows/rust.yml:145]: Yea this was only an attempt to fix cargo-msrv

**Comment by The-personified-devil** [comment]
> Why is this PR draft?

There's complete replacements that also solve the clippy issue, so I want to use them instead, see discord for reference

**Pull request closed without merging**